pkgs/by-name/uc/ucc/package.nix

```diff
@@ -33,43 +56,55 @@ stdenv.mkDerivation rec {
   enableParallelBuilding = true;
 
   postPatch = ''
-
     for comp in $(find src/components -name Makefile.am); do
-      substituteInPlace $comp \
-        --replace "/bin/bash" "${stdenv.shell}"
+      if ! grep -q "/bin/bash" "$comp"; then
+        continue
+      fi
+      substituteInPlace "$comp" \
+        --replace-fail \
+          "/bin/bash" \
+          "${effectiveStdenv.shell}"
     done
   '';
 
   nativeBuildInputs = [
-    libtool
-    automake
     autoconf
-  ] ++ lib.optionals enableCuda [ cudaPackages.cuda_nvcc ];
+    automake
+    libtool
+  ] ++ optionals enableCuda [ cuda_nvcc ];
+
   buildInputs =
     [ ucx ]
-    ++ lib.optionals enableCuda [
-      cudaPackages.cuda_cccl
-      cudaPackages.cuda_cudart
+    ++ optionals enableCuda [
+      cuda_cccl
+      cuda_cudart
+      cuda_nvml_dev
+      nccl
     ];
 
   preConfigure =
+    # Fake libnvidia-ml.so (the real one is deployed impurely)
+    optionalString enableCuda ''
+      addToSearchPath LDFLAGS "-L${getLib cuda_nvml_dev}/lib/stubs"
```

Comment:
H'm, I wonder who's responsible for picking up the real `LDFLAGS`. It's clear that `NIX_LDFLAGS_COMPILE` is picked up by the wrapper, but also we seem to prefer configuring higher-level tools (e.g. cmake) instead of the wrapper when possible

Reply:
I also tried exporting it through `LD_LIBRARY_PATH` but was unsuccessful.
I didn’t try modifying Nix’s equivalent `LDFLAGS` variables, would that be worth a try?
Stubs currently aren’t picked up by most hooks because they’re nested inside the `stubs` subdirectory of `lib`. In the work I’ve done out of tree I add symlinks in the stub output’s `lib` directory to the `stubs` subdirectory so they’re properly detected.

Reply:
> I also tried exporting it through LD_LIBRARY_PATH but was unsuccessful.

That's different, that would need to be set at runtime. Like what we sometimes do with `makeWrapper`. `NIX_LDFLAGS_COMPILE` (Hmm I also see mentions of just `NIX_LDFLAGS` in the repo, which I didn't know exists) is essentially equivalent to `LDFLAGS` but I think it's translated into cli flags by the wrapper. So the 2 questions I have here are:

1. How do we choose between `{,NIX_}LDFLAGS{,_COMPILE}`
2. Why `addToSearchPath` instead of the initial value in `env` attributes

Reply:
I don't have an answer for 1) :(

For 2), I did it because I was unsure of whether setting `env.LDFLAGS` would clobber environment variables modified by setup hooks. Hopefully if there are any that touch it or use it, they're smart enough to not destroy original values.

Reply:
I think not; `env` is like initial values, I think the "builder" program starts in an environment where these are already set: https://github.com/NixOS/nix/blob/442a2623e48357ff72c77bb11cf2cf06d94d2f90/src/libstore/unix/build/local-derivation-goal.cc#L2275

The setup-hooks are run by the "builder", inside the "build", and they normally use `addXXXX` in the expectation that there will be more hooks modifying the same variables before or after?

Reply:
Ah, I think I understand now!

As a secondary concern, are there setup hooks which don't use the `add*` family of bash functions? They could do bad things to environment variables which already have values.